grails-app/views/stockMovement/_events.gsp

```diff
@@ -0,0 +1,137 @@
+    </g:else>
+</section>
+
+<section class="box">
+    <h2>
+        <img src="${resource(dir: 'images/icons/silk', file: 'add.png')}" alt="event" style="vertical-align: middle"/>
+        <label>
+            <g:message code="default.addCustomEvent.label" default="Add Custom Event"/>
+        </label>
+    </h2>
+    <g:form controller="shipment" action="saveEvent" method="POST">
+        <g:hiddenField name="shipmentId" value="${shipmentId}" />
+        <g:hiddenField name="createdBy.id" value="${session.user.id}" />
+        <table>
+            <tbody>
+            %{-- Event type picker --}%
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label>
+                        <g:message code="eventType.label" default="Event type"/>
+                    </label>
+                </td>
+                <td valign="top" class="value ${hasErrors(bean: eventInstance, field: 'eventType', 'errors')}">
+                <g:select id="eventType.id"
+                          name="eventType.id"
+                          optionKey="id"
+                          noSelection="['': g.message(code: 'default.selectOne.label')]"
+                          from='${EventType.listCustomEventTypes()}'
+                          optionValue="${{ format.metadata(obj: it) }}"
+                          required="${true}"
+                          class="chzn-select-deselect">
+                </g:select>
+                </td>
+            </tr>
+            %{-- Location picker --}%
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label>
+                        <g:message code="default.location.label" default="Location" />
+                    </label>
+                </td>
+                <td valign="top" class="value ${hasErrors(bean: eventInstance, field: 'location', 'errors')}">
+                    <g:select id="eventLocation.id"
+                              name="eventLocation.id"
+                              noSelection="['': warehouse.message(code: 'default.selectOne.label')]"
+                              from='${Location.list()}'
+                              optionKey="id"
+                              optionValue="name"
+                              class="chzn-select-deselect">
+                    </g:select>
+                </td>
+            </tr>
+            %{-- Event date picker --}%
+            <tr class="prop">
+                <td valign="top" class="name">
+                    <label>
+                        <g:message code="event.eventDate.label" default="Event date"/>
+                    </label>
+                </td>
+                <td valign="top" class="value ${hasErrors(bean: eventInstance, field: 'eventDate', 'errors')}">
+                    <g:datePicker name="eventDate" value="${eventInstance?.eventDate}" precision="minute"/>
```

Comment:
Should this be required? Or is later set with the current datetime if empty?

Reply:
This picker cannot be empty on form submission. It's preset with the current date.